Solutions/Menes.Testing.AspNetCoreSelfHosting/Menes/Testing/AspNetCoreSelfHosting/OpenApiWebHostManager.cs

```diff
@@ -0,0 +1,122 @@
+    /// Manages the execution of OpenApi services being run as part of a test scenario or feature.
+    /// </summary>
+    /// <remarks>
+    /// <para>
+    /// This class allows you to use the same Startup class that's used in an Azure function to run services in memory.
+    /// To run the same services in memory as your function, obtain an instance of this class (if you wish to run multiple
+    /// services, you can create a single instance of this class to manage all services). Then use the
+    /// <see cref="StartHostAsync"/> method to add services, supplying the Startup class from your function and/or a
+    /// callback to configure the service collection and the base url (including port number where necessary) that the
+    /// endpoints should be made available on.
+    /// </para>
+    /// <para>
+    /// Note that this assumes you're using an instance method in your function host rather than the older static method
+    /// approach. This means that your initialisation will have been moved into a Startup class that implements
+    /// <c>IWebJobsStartup</c> or <c>FunctionsStartup</c>; this is the class that should be supplied to
+    /// <see cref="StartHostAsync"/>.
+    /// </para>
+    /// <para>
+    /// You will normally make this call in a method tagged with either <c>BeforeScenario</c> or <c>BeforeFeature</c>. In the
+    /// corresponding <c>AfterScenario</c> or <c>AfterFeature</c> method, you should call the
+    /// <see cref="StopAllHostsAsync"/> method to stop and dispose the services. If you don't do this, the service will
+    /// remain running. This may impact other features/scenarios as they will not be able to start a new instance of the
+    /// service on the same Url.
+    /// </para>
+    /// </remarks>
+    public class OpenApiWebHostManager
+    {
+        private readonly List<IWebHost> webHosts = new List<IWebHost>();
+
+        /// <summary>
+        /// Starts a new function host using the given Uri and startup class.
+        /// </summary>
+        /// <typeparam name="TFunctionStartup">The type of the startup class. This should be the type of the class from the
+        /// function host project that is used to initialise the OpenApi services and dependencies.</typeparam>
+        /// <param name="baseUrl">The url that the function will be exposed on.</param>
+        /// <param name="additionalServiceConfigurationCallback">
+        /// A callback that will allow you to make changes to the <see cref="IServiceCollection"/> after the code in
+        /// your startup class has executed. You can use this to swap out services for stubs or fakes.
+        /// </param>
+        /// <returns>A <see cref="Task"/> representing the asynchronous operation.</returns>
+        public Task StartHostAsync<TFunctionStartup>(
+            string baseUrl,
+            Action<IServiceCollection>? additionalServiceConfigurationCallback = null)
+            where TFunctionStartup : IWebJobsStartup, new()
```

Comment:
So earlier in this PR you were updating everything from `IWebJobsStartup` to `FunctionsStartup` but not here.

The `IConfiguration`-at-service-init feature we've been waiting for for ages is going to be available via the functions startup stuff. (It's technically also available through the webjobs mechanism, but you have to implement `IWebJobsStartup2` IIRC, and it's clunkier.)

So do we want to move this over to that same world?

Reply:
I left it as `IWebJobsStartup` because with the current state of things, functions can use the newer `FunctionsStartup` or the older combination of `WebJobsStartup` and `IWebJobsStartup`. This way it works for both (since `FunctionsStartup implements IWebJobsStartup`.